stixcore/soc/manager.py

```diff
@@ -0,0 +1,86 @@
+import os
+from pathlib import Path
+from binascii import unhexlify
+from xml.etree import ElementTree as Et
+from collections import defaultdict
+
+from stixcore.tmtc.packets import TMTC, TMPacket
+from stixcore.util.logging import get_logger
+
+__all__ = ['SOCManager', 'SOCPacketFile']
+
+
+logger = get_logger(__name__)
+
+
+class SOCPacketFile:
+    def __init__(self, file):
+        self.file = Path(file)
+        if not self.file.exists():
+            raise ValueError(f"file path not found: {file}")
+        stat = self.file.stat()
+        self.size = stat.st_size
+        self.file_date = stat.st_ctime
+        self.spacecraft_date = None  # TODO some file name conventions?
+        self.tmtc = TMTC.All
+        self.packet_data = defaultdict(list)
+
+        with open(self.file, "r") as f:
```

Comment:
`with self.file.open('r') as f`

Reply:
can search for tags in the xml file e.g `tree.findall('*PktRawResponse')`

Reply:
yes but this takes all or nothing. annd the xml parsing needs quite some time. here at this stage it is just some "security" layer - to check if the content of the xml is "somehow" ok (as expected)
the entire file is not processed at this stage